src/rateLimiters/slidingWindowCounter.ts

```diff
@@ -0,0 +1,95 @@
+import Redis from 'ioredis';
+import { RateLimiter, RateLimiterResponse, RedisWindow } from '../@types/rateLimit';
+
+/**
+ * The SlidingWindowCounter instance of a RateLimiter limits requests based on a unique user ID.
+ * This algorithm improves upon the FixedWindowCounter because this algorithm prevents fixed window's
+ * flaw of allowing doubled capacity requests when hugging the window's borders with a rolling window,
+ * allowing us to average the requests between both windows proportionately with the rolling window's
+ * takeup in each.
+ *
+ * Whenever a user makes a request the following steps are performed:
+ *  1. Fixed minute windows are defined along with redis caches if previously undefined.
```

Comment:
Since window size is configurable we should remove references to minutes in the doc string

Reply:
check